packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -606,3 +614,21 @@ def _init_context_providers(self):
                 **context_providers_kwargs
             )
             self.log.info(f"Registered context provider `{context_provider.id}`.")
+
+    def _init_persona_manager(self, ychat: YChat) -> PersonaManager:
+        config_manager = self.settings["jai_config_manager"]
+        assert config_manager and isinstance(config_manager, ConfigManager)
+
+        try:
+            persona_manager = PersonaManager(
+                ychat=ychat,
+                config_manager=config_manager,
+                event_loop=self.event_loop,
+                log=self.log,
+            )
+        except Exception as e:
+            # TODO: how to stop the extension when this fails
+            # also why do uncaught exceptions produce an empty error log in Jupyter Server?
+            self.log.exception(e)
+
+        return persona_manager
```

Comment:
question: doesn't the `except` clause on line 629 effectively changes the return typing to `PersonaManager | None` ?

Reply:
Nice catch. Fixed this in the latest commit. 👍